### PR TITLE
updated new circuit files and DQN Method

### PR DESCRIPTION
**Pull request by JuanerW**
assemble files for quantum circuit education in RL

### CODE REVIEW

File: Quantum_Circuit_Design/helloworld_DQN_single_file.py [hunk 1]
```diff
@@ -684,17 +667,14 @@ def train_agent(args: Config):
     else:
         buffer = []
     '''start training'''
-    while True:
+    for i in range(100):
```
Comment (Allan-Feng) [inline Quantum_Circuit_Design/helloworld_DQN_single_file.py:670]: Might need better terminal condition without evaluator

File: Quantum_Circuit_Design/helloworld_DQN_single_file.py [hunk 2]
```diff
@@ -716,11 +696,16 @@ def valid_agent(env_class, env_args: dict, net_dims: List[int], agent_class, act
...
     env_class = QuantumEnv
+    target_circuit = text_circuit1
+    num_qubits = 3
+    
     env_args = {
         'env_name': 'Quantum-v1',  # Custom environment name
-        'state_dim': 16,  # QuantumEnv outputs a flattened unitary of size 2 * 4 (assuming 2 qubits)
-        'action_dim':14,  # Number of possible quantum actions (gates)
+        'state_dim': 4**num_qubits,  # QuantumEnv outputs a flattened unitary of size 2 * 4 (assuming 2 qubits)
+        'action_dim':2*num_qubits+num_qubits*(num_qubits-1),  # Number of possible quantum actions (gates)
```
Comment (Allan-Feng) [inline Quantum_Circuit_Design/helloworld_DQN_single_file.py:705]: Currently action dimension need to be changed manually for each universal gate set. Requires more change for automatically calculated action dimension

File: Quantum_Circuit_Design/dqn.py [hunk 3]
```diff
@@ -115,7 +114,6 @@ def _reward(self, target_unitary):
 
         reward = -5 * self.circuit.size()
         done = False
-        reward += fidelity
```
Comment (Chriun) [inline Quantum_Circuit_Design/dqn.py:118]: Why was this removed?

File: Quantum_Circuit_Design/dqn.py [hunk 4]
```diff
@@ -74,30 +73,30 @@ def _circuit_to_state(self, circuit: QuantumCircuit) -> np.ndarray:
...
-            print(len(unitary),len(unitary_array))
-            return unitary
+            return unitary_array
 
     def reset(self):
         self.circuit = QuantumCircuit(self.num_qubits)
         return self._circuit_to_state(self.circuit)
 
+    def generate_actions(self,num_qubits):
+        gates = [HGate(), SGate()]  # Single-qubit gates
```
Comment (Chriun) [inline Quantum_Circuit_Design/dqn.py:83]: Is there no need for the rest of the previously hardcoded gates (TGate(), XXGate(), YGate(), ZGate())?

File: Quantum_Circuit_Design/helloworld_DQN_single_file.py [hunk 5]
```diff
@@ -716,11 +696,16 @@ def valid_agent(env_class, env_args: dict, net_dims: List[int], agent_class, act
...
     # Replace env_class with QuantumEnv
     env_class = QuantumEnv
+    target_circuit = text_circuit1
+    num_qubits = 3
+    
     env_args = {
         'env_name': 'Quantum-v1',  # Custom environment name
-        'state_dim': 16,  # QuantumEnv outputs a flattened unitary of size 2 * 4 (assuming 2 qubits)
-        'action_dim':14,  # Number of possible quantum actions (gates)
+        'state_dim': 4**num_qubits,  # QuantumEnv outputs a flattened unitary of size 2 * 4 (assuming 2 qubits)
```
Comment (Chriun) [inline Quantum_Circuit_Design/helloworld_DQN_single_file.py:704]: State dimension does not seem right. From what I understand, matrices scale in dimension by 2^num_qubits.
